rancher2/config.go

```diff
@@ -1111,16 +1139,57 @@ func (c *Config) UpdateClusterByID(cluster *managementClient.Cluster, update map
 }
 
 func (c *Config) isClusterActive(id string) (bool, *managementClient.Cluster, error) {
-	clus, err := c.GetClusterByID(id)
+	obj, err := c.GetClusterByID(id)
 	if err != nil {
 		return false, nil, err
 	}
 
-	if clus.State == "active" {
-		return true, clus, nil
+	for i := range obj.Conditions {
+		if obj.Conditions[i].Type == clusterActiveCondition {
+			if obj.Conditions[i].Status == "True" {
+				return true, obj, nil
+			}
+			return false, obj, nil
+		}
+	}
+
+	return false, obj, nil
+}
+
+func (c *Config) isClusterMonitoringEnabled(id string) (bool, *managementClient.Cluster, error) {
```

Comment:
Not really sure about this naming, this check if cluster monitoring is Active right?

Reply:
yeah, this is a question how cluster state is checked. This function is checking if cluster condition, `clusterMonitoringEnabledCondition = "MonitoringEnabled"` is true

Reply:
Can we change the function name to include condition so its clear it is checking the condition being enabled and not the config cluster monitoring? Or does that not make sense?

Reply:
Updated function name to `isClusterMonitoringEnabledCondition`